In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

In [2]:
df = pd.read_csv("../data/processed/feature_engineered_data.csv")
df.head(2)

,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,PaperlessBilling,MonthlyCharges,TotalCharges,Churn,...,TechSupport_Yes,StreamingTV_No internet service,StreamingTV_Yes,StreamingMovies_No internet service,StreamingMovies_Yes,Contract_One year,Contract_Two year,PaymentMethod_Credit card (automatic),PaymentMethod_Electronic check,PaymentMethod_Mailed check
0,1,0,1,0,1,0,1,29.85,29.85,0,...,0,0,0,0,0,0,0,0,1,0
1,0,0,0,0,34,1,0,56.95,1889.50,0,...,0,0,0,0,0,1,0,0,0,1


In [3]:
df.shape

(7032, 31)

**Observation:** The feature-engineered dataset contains 7,032 customer records and 31 numerical features ready for model preparation.

In [4]:
# Separating Features and Target
X = df.drop("Churn", axis=1)
y = df["Churn"]

In [5]:
print("X shape:", X.shape)
print("y shape:", y.shape)

X shape: (7032, 30)
y shape: (7032,)


*The independent variables were stored in `X`, while the target variable (`Churn`) was stored in `y` for supervised machine learning.*

In [6]:
# Train-Test Split
X_train, X_test, y_train,y_test = train_test_split(
    X,
    y,
    test_size=0.20,
    random_state=42,
    stratify=y
)

In [7]:
print("X_train:", X_train.shape)
print("X_test :", X_test.shape)
print("y_train:", y_train.shape)
print("y_test :", y_test.shape)

X_train: (5625, 30)
X_test : (1407, 30)
y_train: (5625,)
y_test : (1407,)


*The dataset was divided into training and testing sets using an 80:20 split. Stratified sampling was applied to preserve the original churn distribution in both datasets.*

In [8]:
# Feature Scaling
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [9]:
print(X_train_scaled.shape)
print(X_test_scaled.shape)

(5625, 30)
(1407, 30)


*Standardization was applied to numerical features using `StandardScaler`. The scaler was fitted only on the training data and then applied to the test data to avoid data leakage.*

In [10]:
# Save prepared data
pd.DataFrame(X_train_scaled).to_csv(
    "../data/processed/X_train_scaled.csv",
    index=False
)
pd.DataFrame(X_test_scaled).to_csv(
    "../data/processed/X_test_scaled.csv",
    index=False
)
y_train.to_csv(
    "../data/processed/y_train.csv",
    index=False
)
y_test.to_csv(
    "../data/processed/y_test.csv",
    index=False
)
print("Prepared datasets saved successfully!")

Prepared datasets saved successfully!


# Key Takeaways

* Prepared a machine-learning-ready dataset containing **7,032 customer records and 31 features**.
* Defined **features (`X`)** and the **target variable (`y`)** for churn prediction.
* Performed an **80:20 stratified train-test split** to preserve the original churn distribution.
* Applied **feature scaling using StandardScaler** to standardize numerical features.
* Prevented **data leakage** by fitting the scaler only on the training dataset.
* Generated the final **training and testing datasets** for churn model development and evaluation.
